testdata/quotes.zsh

```diff
@@ -0,0 +1,13 @@
+# Test quoting.
+
+ 'single'   'var: $var'   'subst: $(ls)'   'esc: \n \x01 \001'
+ "double"   "var: $var"   "subst: $(ls)"   "esc: \n \x01 \001" # TODO: these last 2 are not highlighted
```

Comment:
On my machine, without this PR or other recent changes:

- The `$(` and `)` inside double quotes _are_ highlighted, contrary to what the comment says.

- The `\n` `\x01` `\000` _are_ highlighted, contrary to what the comment says… but they shouldn't be:

    ```zsh
    % print -r -- "\n \x01 \001" 
    \n \x01 \001
    ```